nvflare/app_common/executors/statistics/statistics_executor.py

```diff
@@ -0,0 +1,316 @@
+        client_name = fl_ctx.get_prop(ReservedKey.CLIENT_NAME)
+        self.log_info(fl_ctx, f"Executing task '{task_name}' for client: '{client_name}'")
+        try:
+            result = self.client_exec(task_name, shareable, fl_ctx, abort_signal)
+            if abort_signal.triggered:
+                return make_reply(ReturnCode.TASK_ABORTED)
+            if result:
+                dxo = DXO(data_kind=DataKind.ANALYTIC, data=result)
+                return dxo.to_shareable()
+            else:
+                return make_reply(ReturnCode.EXECUTION_EXCEPTION)
+
+        except BaseException as e:
+            self.log_exception(fl_ctx, f"Task {task_name} failed. Exception: {e.__str__()}")
+            return make_reply(ReturnCode.EXECUTION_EXCEPTION)
+
+    def client_exec(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
+        client_name = fl_ctx.get_prop(ReservedKey.CLIENT_NAME)
+        self.log_info(fl_ctx, f"Executing task '{task_name}' for client: '{client_name}'")
+        result = Shareable()
+        metrics_result = {}
+        if task_name == StC.FED_STATS_TASK:
+            ds_features = self.get_numeric_features()
+            self.validate(client_name, ds_features, shareable, fl_ctx)
+            metric_task = shareable.get(StC.METRIC_TASK_KEY)
+            target_metrics: List[MetricConfig] = shareable.get(StC.STATS_TARGET_METRICS)
+            for tm in target_metrics:
+                fn = self.metric_functions()[tm.name]
+                metrics_result[tm.name] = {}
+                for ds_name in ds_features:
+                    metrics_result[tm.name][ds_name] = {}
+                    features: List[Feature] = ds_features[ds_name]
+                    for feature in features:
+                        metrics_result[tm.name][ds_name][feature.feature_name] = fn(
+                            ds_name, feature.feature_name, tm, shareable, fl_ctx
+                        )
+
+            result[StC.METRIC_TASK_KEY] = metric_task
+            if metric_task == StC.STATS_1st_METRICS:
+                result[StC.STATS_FEATURES] = ds_features
+
+            result[metric_task] = metrics_result
+
+        return result
+
+    def get_numeric_features(self) -> Dict[str, List[Feature]]:
+        ds_features: Dict[str, List[Feature]] = self.stats_generator.features()
+        return filter_numeric_features(ds_features)
+
+    def validate(
+        self, client_name: str, ds_features: Dict[str, List[Feature]], shareable: Shareable, fl_ctx: FLContext
+    ):
+        count_config = MetricConfig(StC.STATS_COUNT, {})
+        for ds_name in ds_features:
+            features = ds_features[ds_name]
+            for feature in features:
+                feature_name = feature.feature_name
+                count = self.get_count(ds_name, feature.feature_name, count_config, shareable, fl_ctx)
+                if count < self.min_count:
+                    raise ValidationException(
```

Comment:
suggest just use `StatisticsExecutorException` or something similar rather than make this a generic name, we could unify some of these exceptions later.